actions/utils.js

```diff
@@ -0,0 +1,61 @@
+export class Action {
+    constructor(normandy, recipe) {
+        this.normandy = normandy;
+        this.recipe = recipe;
+    }
+}
+
+/**
+ * From the given list of objects, choose one based on their relative
+ * weights and return it. Choices are assumed to be objects with a `weight`
+ * property that is an integer.
+ *
+ * Weights define the probability a choices will be shown relative to other
+ * weighted choices. If two choices have weights 10 and 20, the second one will
+ * appear twice as often as the first.
+ *
+ * @param  {array}  choices  Array of weighted choices.
+ * @return {object}          The chosen choice.
+ */
+export function weightedChoose(choices) {
+    // Rolling sums of weights; each weight is the choice's weight plus
+    // the weight of all previous choices.
+    let weights = [choices[0].weight];
+    for (let choice of choices.slice(1)) {
+        weights.push(choice.weight + weights[weights.length - 1]);
+    }
+
+    let choice = Math.random() * weights[weights.length - 1];
```

Comment:
I would maybe change the name of this variable from `choice` to something like...`weightLimit`? `chosenWeight`? `randomWeight`?

It's a little hard to follow this code with the current naming...`choices` represents an array of objects that each have a weight property, but `choice` is a random integer weight we're using to determine `chosen`. (Unless you're inside the for loop on line 24, in which case `choice` is an object with a weight property.)